### ISSUE TITLE
SchemaField#default: provide variant that decodes the document internally

### ISSUE THREAD

**Issue by kubukoz** [disneystreaming/smithy4s#382]
Something like 

```
//in implicit class for SchemaField[S, A]
def getDefaultValue: Option[A] = getDefault.map(getCachedDocumentDecoder[A].apply(_))
```

**Comment by Baccata** [comment]
It should go on the Schema as opposed to the field (which is polymorphic), but sure.

**Comment by lewisjkl** [comment]
Will consider this closed once I add a PR that utilizes my changes from #580

**Comment by kubukoz** [comment]
Closed in #588
